src/hotspot/cpu/aarch64/interp_masm_aarch64.cpp

```diff
@@ -1399,8 +1400,15 @@ void InterpreterMacroAssembler::_interp_verify_oop(Register reg, TosState state,
 
 void InterpreterMacroAssembler::verify_FPU(int stack_depth, TosState state) { ; }
 
-void InterpreterMacroAssembler::end_training_check() {
-  call_VM(noreg, CAST_FROM_FN_PTR(address, InterpreterRuntime::end_training_check));
+void InterpreterMacroAssembler::generate_runtime_upcalls_on_method_entry()
+{
+  address upcall = RuntimeUpcalls::on_method_entry_upcall_address();
+  if (RuntimeUpcalls::does_upcall_need_method_parameter(upcall)) {
+    get_method(c_rarg1);
+    call_VM(noreg,upcall, c_rarg1);
```

Comment:
Does call_VM actually make a call, or does it set up a call to be made later? If it's the former, I'm not sure why "generate" is in the name of this method (it should be something like "execute").

Reply:
It generates the code to make the call for when the compiled method is executed later